### DISCUSSION TITLE
Add hardware offload support for 5.9.6 version

### DISCUSSION THREAD

**Discussion by ferasbi** [strongswan/strongswan#1109] [General]
Hi,

I'm trying to add hardware offload support for strongswan 5.9.6. I'm trying to do something similar to what was done here:
https://github.com/Mellanox/strongswan on the BF-5.9.0 branch.
I made changes on the 5.9.6 branch exactly like the changes that were done BF-5.9.0 branch from the 5.9.0 branch, except for the changes on /src/libstrongswan/plugins/openssl/openssl_ec_diffie_hellman.c (which changed to use ECDH_compute_key function) because I saw that compute_shared_key() function was dropped in 5.9.6 and now you use a different way of computing the shared key. I'm trying to use the ECDH_compute_key() function like they did in BF-5.9.0 version but I don't know where to integrate it. Is ECDH_compute_key() really needed for HW offload? If so, where do you think I can integrate it? I will be very grateful to hear any suggestion and thoughts on adding HW offload to strongswan 5.9.6 (which will be used on network devices that support hardware offloading of IPsec).

Kind regards,
Feras

**Comment by tobiasbrunner** [comment]
>  Is ECDH_compute_key() really needed for HW offload?

No, that has nothing to do with hardware offloading of IPsec (ECDH may be used for key derivation when creating IKE or IPsec SAs).
